pkg/agent/entrypoint.go

```diff
@@ -88,6 +88,10 @@ func (r MainRunner) Run(logger *zap.Logger, ctx context.Context) error {
 	}
 
 	tg := taskgroup.NewGroup(logger, taskgroup.WithParentContext(ctx))
+	tg.Go("scalingevents-run", func(logger *zap.Logger) error {
```

Comment:
One thought I had: how do we handle panics? ideally, a panic from this task (or, any of the tasks in its own taskgroup) triggers shutdown of the autoscaler-agent.

We can set the top-level taskgroup's panic handler to cancel the parent context, but doing that from the child taskgroup is a little trickier.

Going to think about this, but cc @Omrigan in case you have thoughts.

Reply:
UPD: `tg.Ctx()` can be used here as well - just return the error from the task when you want for the entire group to finish. See also https://github.com/neondatabase/autoscaling/pull/1235

Reply:
Makes sense -- Seems like switching to "just" propagating errors is sufficient. Did that with ed73d72b9d1405cc6a99a66cd12e3513f3bf44cd